cleverpush/src/main/java/com/cleverpush/DeviceIdBroadcastReceiver.java

```diff
@@ -0,0 +1,45 @@
+package com.cleverpush;
+
+import static com.cleverpush.CleverPush.context;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+
+import com.cleverpush.util.Logger;
+
+public class DeviceIdBroadcastReceiver extends BroadcastReceiver {
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        if (intent.getStringExtra(Constants.GET_FULL_PACKAGE_NAME_KEY).equals(Constants.APPLICATION_PACKAGE_NAME)) {
+            return;
+        }
+
+        if (intent.getAction().equals(Constants.GET_DEVICE_ID_FROM_ALL_DEVICE)) {
+            sendBrodCastReceiver();
+        }
+
+        if (intent.getAction().equals(Constants.DEVICE_ID_ACTION_KEY) && intent.hasExtra(Constants.DEVICE_ID)) {
+            SharedPreferences sharedPreferences = getSharedPreferences(context);
+            String deviceId = intent.getStringExtra(Constants.DEVICE_ID);
+            sharedPreferences.edit().putString(CleverPushPreferences.DEVICE_ID, deviceId).apply();
+        }
+
+    }
+
+    private void sendBrodCastReceiver() {
```

Comment:
DEVICE_ID_ACTION_KEY missing for the intent?

Reply:
Added